core/shared/src/main/scala/kantan/csv/HeaderDecoder.scala

```diff
@@ -25,9 +25,33 @@ package kantan.csv
   * available), but you can create more useful [[HeaderDecoder]] instances through the
   * [[HeaderDecoder$ companion object]].
   */
-trait HeaderDecoder[A] extends Serializable {
+trait HeaderDecoder[A] extends Serializable { self =>
   def fromHeader(header: Seq[String]): DecodeResult[RowDecoder[A]]
   def noHeader: RowDecoder[A]
+
+  /**
+    * Combines two header decoders creating a tupled version that will decode the results from both. The combination will preserve the order
+    * of the merge in the case that no headers are detected.
+    */
+  def ~[B](that: HeaderDecoder[B])(implicit zippable: Zippable[A, B]): HeaderDecoder[zippable.Out] =
```

Comment:
I chose this name just because it is used in alot of other parsing libraries, but I can change to a non-symbolic operator (or add this simply as an alias) if that is preferred.